### PR TITLE
Use public calendar key for subscriptions only

### PR DESCRIPTION
**Pull request by johnnyporkchops**
## Summary (required)
Use `FEC_API_KEY_PUBLIC_CALENDAR` for Calendar subscriptions and use `FEC_API_KEY_PUBLIC` for homepage Event and deadlines links and Calendar filters. 

- Resolves https://github.com/fecgov/fec-accounts/issues/354

## Impacted areas of the application
modified:  fec/fec/static/js/modules/calendar-helpers.js
modified:  fec/fec/static/js/modules/calendar.js
modified:  fec/fec/static/js/pages/calendar-page.js
modified:  fec/fec/static/js/templates/calendar/subscribe.hbs
modified:  fec/fec/tests/js/calendar.js

## Screenshots

cal-api

## How to test
checkout feature/345-use-calendar-key-for-subscriptions-only
`npm run build`
`export FEC_WEB_API_KEY_PUBLIC=12345`
`export FEC_WEB_API_KEY_PUBLIC_CALENDAR=67890`
runserver
Check that the links in subscribe dropdown have  `api_key=67890`
Check in network tab that changes to filters register an api call using `api_key=12345`
Check that clicks from links on homepage Event and deadlines section (except Meetings) register an api call using `api_key=12345` in network tab.

____

### CODE REVIEW

**Review by pkfec — APPROVED** [review]
Thanks @johnnyporkchops for working on this issue and fixing it so quickly.  Works as expected.

File: fec/fec/tests/js/calendar.js [hunk 1]
```diff
@@ -282,10 +283,30 @@ describe('helpers', function() {
...
+    it('builds the correct subscribe url', function() {
+      var subscribeUrl = calendarHelpers.getUrl('calendar', {category: 'election'}, 'sub');
+      expect(subscribeUrl).to.equal('/v1/calendar/?api_key=67890&per_page=500&category=election');
+    
+    });
+
   });
 
+  describe('calendar.updateLinks()', function() {
+      it('builds the correct, decooded googlerSubscribe url', function() {
```
Comment (lbeaufort) [inline fec/fec/tests/js/calendar.js:298]: I think this should say `decoded googleSubscribe url`

**Review by lbeaufort — APPROVED** [review]
Looks great, thanks @johnnyporkchops! Thanks for adding test coverage - just a small comment on one of the tests

File: fec/fec/tests/js/calendar.js (hunk 1 above)
Comment (johnnyporkchops) [reply fec/fec/tests/js/calendar.js]: Thanks Laura, I changed it to encoded which I think is more accurate

Comment (lbeaufort) [reply fec/fec/tests/js/calendar.js]: Sounds good @johnnyporkchops! Also `googlerSubscribe` should be `googleSubscribe`, right?

**Pull request merged**